plugins/libplugin-pay.c

```diff
@@ -453,57 +468,88 @@ payment_constraints_update(struct payment_constraints *cons,
  * through, since the balances really changed in that case. The `remove`
  * argument indicates whether we want to apply (`remove=false`), or clear a
  * prior application (`remove=true`). */
-static void payment_chanhints_apply_route(struct payment *p, bool remove)
+static bool payment_chanhints_apply_route(struct payment *p, bool remove)
 {
+	bool apply;
 	struct route_hop *curhop;
 	struct channel_hint *curhint;
 	struct payment *root = payment_root(p);
 	assert(p->route != NULL);
+
+	/* First round: make sure we can cleanly apply the update. */
 	for (size_t i = 0; i < tal_count(p->route); i++) {
+		/* No need to check for applicability if we increase
+		 * capacity and budgets. */
+		if (remove)
+			continue;
```

Comment:
The function argument `remove` is never written to within this function. Thus, if `remove` is `false` it can be expected that this`if (remove) continue;` statement leads to a pointless(?) iteration through this route and it would be the same as `if (remove) break;` just a little slower. Is this intended?

Reply:
I know, I just didn't want to indent the whole `for`-loop into an `if` :-)

Reply:
then why not `break` instead of `continue`? But not that important ;)